src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -247,7 +291,39 @@ public void RemoveItems(IEnumerable collection)
         /// <returns>IDisposable.</returns>
         public IDisposable SuspendChangeNotifications()
         {
-            return new DisposableToken<FastObservableCollection<T>>(this,
+            return SuspendChangeNotifications(SuspensionMode.Mixed);
+        }
+
+        /// <summary>
+        /// Suspends the change notifications until the returned <see cref="IDisposable"/> is disposed.
+        /// <example>
+        /// <code>
+        /// <![CDATA[
+        /// var fastCollection = new FastObservableCollection<int>();
+        /// using (fastCollection.SuspendChangeNotificaftions())
+        /// {
+        ///     // Adding or removing events inside here will not raise events
+        ///     fastCollection.Add(1);
+        ///     fastCollection.Add(2);
+        ///     fastCollection.Add(3);
+        /// 
+        ///     fastCollection.Remove(3);
+        ///     fastCollection.Remove(2);
+        ///     fastCollection.Remove(1);
+        /// }
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="mode">The suspension Mode.</param>
+        /// <returns>IDisposable.</returns>
+        public IDisposable SuspendChangeNotifications(SuspensionMode mode)
+        {
+            // Set suspension mode
+            _suspensionMode = mode;
```

Comment:
I think storing the suspension mode should be inside the disposable token, but we might even store it in the SuspensionContext.